### PR TITLE
[Merged by Bors] - chore: removes non-breaking spaces (U+00a0)

### PR DESCRIPTION
**Pull request by adomasbaliuka**
Removes all non-breaking spaces (U+00a0) from Mathlib.

This is in preparation for using the "unicode linter" (#16215), which would (as of now) disallow non-printing characters except for space (U+0020) and linefeed (U+000A).

---

Relevant discussions:

- [Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Whitelist.20for.20Unicode.3F/near/448721083) 

- #16215 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Are there any such spaces in the archive, tests or counterexamples?
(This need not block this PR; I think these changes are useful in any case.)

File: Mathlib/Condensed/Light/TopCatAdjunction.lean [hunk 1]
```diff
@@ -73,7 +73,7 @@ def toTopCatMap : X.toTopCat ⟶ Y.toTopCat where
     rw [this]
     exact continuous_coinducingCoprod _ _
 
-/-- The functor `LightCondSet ⥤ TopCat`  -/
+/-- The functor `LightCondSet ⥤ TopCat`  -/
```
Comment (grunweg) [inline Mathlib/Condensed/Light/TopCatAdjunction.lean:76]:
```suggestion
/-- The functor `LightCondSet ⥤ TopCat` -/

Comment (grunweg) [reply Mathlib/Condensed/Light/TopCatAdjunction.lean:76]: Optional, here and above

**Review by grunweg — COMMENTED** [review]
Thanks!
maintainer merge

**Comment by adomasbaliuka** [comment]
> Are there any such spaces in the archive, tests or counterexamples? (This need not block this PR; I think these changes are useful in any case.)

As far as I can see, there are no more in the entire repository.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Comment by adomasbaliuka** [comment]
@sgouezel it seems that a linter complains about a too long line in a file this PR didn't change? How can that happen?

**Comment by adomasbaliuka** [comment]
@grunweg @sgouezel the "overlong" line that the bors build complains about isn't actually overlong. So I have no idea what's happening here...

**Comment by bryangingechen** [comment]
The lint failure might be from another PR currently being merged or from a commit in `master` which isn't in this branch yet. In any case this PR now has a merge conflict with `master` as well, so I'm taking it off the queue:
bors r-

After merging `master` and fixing any errors, feel free to put this back on the queue:
bors d+

**Pull request closed without merging**